packages/osc-ui/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,275 @@
+import type {
+    AutocompleteOptions,
+    AutocompleteSource,
+    AutocompleteState,
+} from '@algolia/autocomplete-core';
+import { createAutocomplete } from '@algolia/autocomplete-core';
+import { createAlgoliaInsightsPlugin } from '@algolia/autocomplete-plugin-algolia-insights';
+import { getAlgoliaResults } from '@algolia/autocomplete-preset-algolia';
+import type { Hit } from '@algolia/client-search';
+import algoliasearch from 'algoliasearch/lite';
+import type { BaseSyntheticEvent, KeyboardEvent, MouseEvent } from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
```

Comment:
Can we change the import to the remix version? 
```suggestion
import { useNavigate } from '@remix-run/react';
```

`react-router-dom` is installed but only as a dev dependency for using the `MemoryRouter` in tests.